src/hipscat_import/cross_match/macauff_metadata.py

```diff
@@ -0,0 +1,102 @@
+"""Utilities for generating parquet metdata from macauff-generated metadata files."""
+
+import xml.etree.ElementTree as ET
+
+import pyarrow as pa
+import pyarrow.parquet as pq
+import yaml
+from hipscat.io import file_io
+
+
+def _get_inner_xml_value(parent_el, node_type, default_value):
+    child_el = parent_el.findall(node_type)
+    if len(child_el) == 0:
+        return default_value
+    if len(child_el) > 1:
+        raise ValueError(f"found too many {node_type} XML elements")
+    return child_el[0].text.strip()
+
+
+def _construct_field(name, units, metadata_dict):
+    """Helper method to construct a pyarrow field from macauff metadata strings."""
+    if units == "string":
+        pa_type = pa.string()
+    elif units in ("float", "double"):
+        pa_type = pa.float64()
+    elif units in ("integer", "long"):
+        pa_type = pa.int64()
+    else:
+        raise ValueError(f"unhandled units {units}")
+    return pa.field(name, pa_type, metadata=metadata_dict)
+
+
+def from_xml(input_file, output_file):
```

Comment:
I think this is fine to leave in for now but afaik from Tom we'll only have to read in YAML files from here on out so we should consider whether we need the XML functionality in a few weeks once we have everything working as intended with the YAML files.